### PR TITLE
Derive zio.test.Gen

### PR DESCRIPTION
**Pull request by igrebenik**
My second attempt on resolving #56. 
I've renewed this pr due to recent changes in zio-schema

### CODE REVIEW

**Comment by igrebenik** [comment]
I'll add the tests a bit later

**Comment by igrebenik** [comment]
@thinkharderdev Hey Dan! Could you have a look at this PR?!

**Comment by igrebenik** [comment]
Could someone explain me why is this happening?
```
[error] /home/runner/work/zio-schema/zio-schema/zio-schema-zio-test/shared/src/main/scala/zio/schema/test/DeriveGen.scala:35:141: unreachable code
[error]       case Schema.Enum2(c1, c2, _)                                                                                               => genOneOf(c1, c2)
[error]                                                                                                                                             ^
[error] /home/runner/work/zio-schema/zio-schema/zio-schema-zio-test/shared/src/main/scala/zio/schema/test/DeriveGen.scala:67:14: unreachable code
[error]           f1 <- gen(field1.schema)
[error]              ^
```
`Schema.CaseClass2` and `Schema.Enum2` ?!

File: zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+            listMap <- genListMap
+            value   <- gen(field.schema)
+          } yield listMap.updated(field.label, value)
+      }
+
+  private def genSequence[Z, A](seq: Schema.Sequence[Z, A]): Gen[Random with Sized, Z] =
+    Gen.chunkOf1(gen(seq.schemaA)).map(nec => seq.fromChunk(nec.toChunk))
+
+  private def genMap[K, V](map: Schema.MapSchema[K, V]): Gen[Random with Sized, Map[K, V]] =
+    Gen.mapOf1(gen(map.ks), gen(map.vs))
```
Comment (thinkharderdev) [inline zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala:468]: For these two I would not restrict to single-element chunks (at the very least we should sometimes generate an empty Chunk). So maybe a chunk bounded by 0 and 2? I know in other places we run into severe performance issues if  we don't restrict to small(ish) collection sized.

**Review by thinkharderdev — COMMENTED** [review]
Looks great! One small comment but I think this is ready to merge otherwise.

File: zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala (hunk 1 above)
Comment (igrebenik) [reply zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala:468]: Yes, I was concerned about chunk's arbitrary size and that's why I did single-element chunks).
No problem, I will bound the chunk by 0 and 2

**Review by thinkharderdev — APPROVED** [review]
Great work!

**Pull request merged**